### ISSUE TITLE
Typing Cleanup - Remove Blacklisted Items

### ISSUE THREAD

**Issue by WillAyd** [pandas-dev/pandas#25882]
#25844 introduced a blacklist of modules in the mypy.ini config file which currently throw errors when analyzed with mypy. We would ideally be able to whittle these down (save exclusion of the test folder(s), which I think is fine).

I'm not sure there is one easy way to split it up as the degree of difficulty may vary wildly. PRs from the community to clean up one to a few of the blacklisted modules at a time would certainly be welcome!

Here's the list of modules for reference:

- [X] pandas._config.config (#25899)
- [X] pandas._version (#25949)
- [X] pandas.compat (#26251)
- [X] pandas.compat.numpy.function
- [X] pandas.compat.pickle_compat (#25991)
- [X] pandas.core.accessor (#26146)
- [x] pandas.core.api
- [X] pandas.core.apply
- [X] pandas.core.arrays.array_  (#26065)
- [X] pandas.core.arrays.datetimelike  (#26065)
- [X] pandas.core.arrays.integer  (#26065)
- [X] pandas.core.arrays.interval (#26065)
- [X] pandas.core.arrays.numpy_ (#26065)
- [X] pandas.core.arrays.period (#26065)
- [X] pandas.core.arrays.sparse (#26065)
- [X] pandas.core.arrays.timedeltas (#26065)
- [x] pandas.core.base
- [x] pandas.core.computation.expr
- [x] pandas.core.computation.ops
- [x] pandas.core.computation.pytables
- [x] pandas.core.config (#26048)
- [x] pandas.core.config_init
- [X] pandas.core.dtypes.dtypes (#26028)
- [X] pandas.core.dtypes.missing (#26028)
- [X] pandas.core.frame
- [X] pandas.core.generic
- [X] pandas.core.groupby.generic (#26089)
- [X] pandas.core.groupby.groupby (#26089)
- [X] pandas.core.groupby.ops (#26089)
- [x] pandas.core.indexes.base (#26315)
- [x] pandas.core.indexes.datetimelike
- [ ] pandas.core.indexes.datetimes (#26315)
- [x] pandas.core.indexes.period (#26315)
- [x] pandas.core.indexes.timedeltas (#26315)
- [x] pandas.core.indexing
- [x] pandas.core.internals.blocks
- [x] pandas.core.ops
- [X] pandas.core.panel (DEPRECATED - do not update)
- [x] pandas.core.resample
- [X] pandas.core.reshape.concat (#26019)
- [x] pandas.core.reshape.merge
- [x] pandas.core.reshape.reshape
- [X] pandas.core.reshape.tile (#26019)
- [X] pandas.core.series
- [X] pandas.core.sparse.frame
- [X] pandas.core.strings (#25902)
- [x] pandas.core.util.hashing
- [x] pandas.core.window
- [x] pandas.io.clipboards (#26048)
- [x] pandas.io.feather_format (#25939)
- [x] pandas.io.formats.css (#25900)
- [x] pandas.io.html (#25939)
- [x] pandas.io.json.json (#25939)
- [x] pandas.io.json.normalize (#25939)
- [x] pandas.io.json.table_schema (#25939)
- [x] pandas.io.packers (#25939)
- [x] pandas.io.parquet (#25939)
- [x] pandas.io.pytables
- [x] pandas.io.stata (#25940)
- [X] pandas.plotting._core (#26021)
- [X] pandas.tseries.frequencies (#26014)
- [X] pandas.tseries.holiday (#26014)
- [X] pandas.tseries.offsets (#26014)
- [x] pandas.util._doctools
- [x] pandas.util.testing

**Comment by GreybeardCG** [comment]
@WillAyd I've been studying a few of these files and related contributions today and was wondering if there are any catches I need to know about when fixing these mypy violations. 
As an example: in array_.py:231 adding dtype = cast(ExtensionDtype, dtype)" seems the least intrusive. However, I saw that typing was [added to this file partially already](https://github.com/pandas-dev/pandas/pull/25829/files) , so was there a specific reason to leave this violation in? Thanks a lot!

**Comment by WillAyd** [comment]
Have you merged master?

**Comment by GreybeardCG** [comment]
@WillAyd Yes, I'm on the latest upstream/master.

**Comment by WillAyd** [comment]
Sorry misread comment. We’ve had a few things on one of which was simply to compat syntax to Py3 annotation

For the module you mentioned we did that but haven’t focused on actually fixing the errors. If you still see some and want to submit a PR would certainly be welcome

**Comment by GreybeardCG** [comment]
@WillAyd All clear then! I'll keep working on it.

**Comment by gwrome** [comment]
@WillAyd I've run into a few cases where we should use an old style type hint for a global variable to maintain pre-3.6 compatibility, but the appropriate hint is just too long for the line. I haven't been able to find any documented way to break a variable's type hint across lines or put it on another line. 

Any ideas?

For example in pandas/compat/numpy/function.py:105 we want something like 
```py
ARGSORT_DEFAULTS = OrderedDict()  # type: OrderedDict[str, Union[int, str, None]]
```
but that line is too long. We could replace the `Union` with `Any`, but that reduces the usefulness of type checking. In this particular case, it probably doesn't matter because that ARGSORT_DEFAULTS  gets fed directly into a classconstructor and is never touched again. But there may be other places in the code where accuracy is more important and the line length is an issue.

**Comment by Naddiseo** [comment]
@gwrome https://github.com/python/mypy/issues/4511 might be of use

tldr: type alias, or backslash at end of line

**Comment by gwrome** [comment]
@WillAyd, I think these are ready for removal from the list above, because they're no longer in the blacklist. I've included the PR if I could find it:

- pandas.compat (#26251)
- pandas.compat.numpy.function
- pandas.core.accessor (#26146)
- pandas.core.apply
- pandas.core.arrays.* (#26065)
- pandas.core.dtypes.* (#26028)
- pandas.core.frame
- pandas.core.generic
- pandas.core.groupby.* (#26089)
- pandas.core.reshape.concat (#26019)
- pandas.core.reshape.tile (#26019)
- pandas.core.series
- pandas.core.sparse.frame

**Comment by WillAyd** [comment]
Great - I’ve been meaning to update this so this helps a lot. 

Thanks for all of the help on these!

**Comment by gwrome** [comment]
You’re welcome. Hoping to knock a couple more out this weekend.
